pkg/metrics/config.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/knative/pkg/metrics"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	ObservabilityConfigName = "config-observability"
+	metricsDomain           = "knative.dev/eventing"
+	defaultLogURLTemplate   = "http://localhost:8001/api/v1/namespaces/knative-monitoring/services/kibana-logging/proxy/app/kibana#/discover?_a=(query:(match:(kubernetes.labels.knative-dev%2FrevisionUID:(query:'${REVISION_UID}',type:phrase))))"
+)
+
+// UpdateExporterFromConfigMap returns a helper func that can be used to update the exporter
+// when a config map is updated
+func UpdateExporterFromConfigMap(component string, logger *zap.SugaredLogger) func(configMap *corev1.ConfigMap) {
+	return func(configMap *corev1.ConfigMap) {
+		metrics.UpdateExporter(metrics.ExporterOptions{
+			Domain:    metricsDomain,
+			Component: component,
+			ConfigMap: configMap.Data,
+		}, logger)
+	}
+}
+
+// ObservabilityConfig contains the configuration defined in the observability ConfigMap.
+type ObservabilityConfig struct {
+	// EnableVarLogCollection dedicates whether to set up a fluentd sidecar to
+	// collect logs under /var/log/.
+	EnableVarLogCollection bool
+
+	// TODO(#818): Use the fluentd daemon set to collect /var/log.
```

Comment:
This seems wrong or should be pointer to another repo?
https://github.com/knative/eventing/issues/818